testing/mock/mock.go

```diff
@@ -89,15 +88,14 @@ func (AppModuleBasic) GetQueryCmd() *cobra.Command {
 // AppModule represents the AppModule for the mock module.
 type AppModule struct {
 	AppModuleBasic
-	scopedKeeper capabilitykeeper.ScopedKeeper
-	portKeeper   PortKeeper
+	ibcApps    []*MockIBCApp
```

Comment:
So this is so that one `AppModule` can be shared between multiple `MockIBCApps`?
I find it slightly confusing how `IBCModule` contains a reference to its `MockIBCApp` but also a reference to `AppModule` which contains references to all other `MockIBCApps`. 

Is there a way it could be structured so that you could call `NewAppModule` per mock application? I guess different args would need to be supplied if that was done and maybe complicate things further. Perhaps I'm overthinking things a bit, I'm just thinking out loud here really for what may be the easiest to follow option.

Reply:
> Is there a way it could be structured so that you could call `NewAppModule` per mock application? 

No this was the purpose of splitting out the IBC callbacks. The alternative approaches:
- developers pass in all IBC apps to `NewAppModule`
- global variable

I felt the dev UX improvements were worth the under the hood magic (hence why those variables are private)

Reply:
> No this was the purpose of splitting out the IBC callbacks

Ah yes, that slipped my mind from before! Got a little lost in the weeds thinking of alternatives. Makes sense along with the context from your other comment too, thanks a lot for the thorough explanation! I agree, since this is kind of hidden away anyway, its totally worth it for the testability of middleware..etc